src/worker.rs

```diff
@@ -250,4 +243,17 @@ impl<T: Sync + Send + 'static> Worker<T> {
             }
         }
     }
+
+    fn prepare_new_matches(&mut self) {
+        self.matches.clear();
+        self.matches.reserve(
+            self.matches
+                .capacity()
+                .saturating_sub(self.last_snapshot as usize),
+        );
```

Comment:
this reserve seem incorrect to me, It should just be `self.last_snapshot as usize`? We clear the matches array and then extend it with `self.last_snapshot` elements

Reply:
`reserve`  is the worst named function in Rust, it takes additional capacity, not total and `self.matches` does not grow later in the function so the final capacity will be `self.last_snapshot`, meaning we only need to add the diff in added capacity.

I still made a mistake by inverting the two 😅

Reply:
Actually your understanding of that function is slightly wrong I believe:

> Reserves capacity for at least **additional more elements to be inserted** in the given Vec<T>

`reserve` makes spaces for `n` additionl elements to be inserted. So calling `push` `n` time afterwards is garunteed to never allocate. In other words after a call to reserve  `capacity >= n + len`. 

Since `len` is `0` in this case you should just use the final length here (so don't subtract capacity).